server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -215,7 +216,12 @@ public ExpressionAnalyzer(Functions functions,
      * Functions with constants will be normalized.
      */
     public Symbol convert(Expression expression, ExpressionAnalysisContext expressionAnalysisContext) {
-        return expression.accept(innerAnalyzer, expressionAnalysisContext);
+        var symbol = expression.accept(innerAnalyzer, expressionAnalysisContext);
+        var normalizer = EvaluatingNormalizer.functionOnlyNormalizer(
+            functions,
+            f -> expressionAnalysisContext.isEagerNormalizationAllowed() && f.info().isDeterministic()
```

Comment:
Would it be possible to remove the `isEagerNormalizationAllowed` flag now altogether? 

We've afaik only 2 places where it is disabled (for using literals as table function and in the values case), in those cases we could call `expression.accept..` directly instead of going through `convert`.

Reply:
Nevermind. I think then it wouldn't apply for inner convert calls. Guess we have to keep it.